client/verta/tests/unit_tests/pipeline/test_pipeline_graph.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+"""
+Unit tests for the PipelineGraph class
+"""
+
+from hypothesis import given, HealthCheck, settings
+
+from tests.unit_tests.strategies import mock_pipeline_definition
+from verta.pipeline import PipelineGraph
+
+
+def test_set_steps(make_mock_pipeline_step) -> None:
+    """
+    Test that the steps of a PipelineGraph can be set
+    """
+    step_1 = make_mock_pipeline_step()
+    step_2 = make_mock_pipeline_step()
+    graph = PipelineGraph(steps=[])
+    graph.set_steps([step_1, step_2])
+    assert set(graph.steps) == set([step_1, step_2])
+    graph.set_steps([])
+    assert not graph.steps
+
+
+@given(pipeline_definition=mock_pipeline_definition())
+@settings(
+    suppress_health_check=[HealthCheck.function_scoped_fixture],
+    deadline=None,
+)
+def test_from_definition(
+    mocked_responses, pipeline_definition, mock_conn, mock_config
+) -> None:
+    """
+    Test that a PipelineGraph object can be constructed from a pipeline
+    specification.  The model version is fetched for each step, so a response
+    is mocked for each.  In depth testing of each step is handled in
+    test_pipeline_step.test_steps_from_pipeline_spec.
+    """
+    for step in pipeline_definition["steps"]:
+        mocked_responses.get(
+            f"https://test_socket/api/v1/registry/model_versions/{step['model_version_id']}",
+            json={"name": "test"},
+            status=200,
+        )
+        mocked_responses.get(
+            f"https://test_socket/api/v1/registry/registered_models/0",
+            json={},
+            status=200,
+        )
+    graph = PipelineGraph._from_definition(
+        pipeline_definition=pipeline_definition, conn=mock_conn, conf=mock_config
+    )
+    assert isinstance(graph, PipelineGraph)
+    assert len(graph.steps) == len(pipeline_definition["steps"])
```

Comment:
Can we do a stronger check than just the number of steps?

Reply:
The tests that mock http calls have been beefed up to provide more realistic json responses, which has allowed more thorough assertions.